packages/bull-vault/src/LeverageBull.sol

```diff
@@ -85,6 +94,42 @@ contract LeverageBull is Ownable {
         transferOwnership(_owner);
     }
 
+    function setAuction(address _auction) external onlyOwner {
+        require(_auction != address(0), "BS3");
+
+        emit SetAuction(auction, _auction);
+
+        auction = _auction;
+    }
+
+    function repayAndWithdrawFromLeverage(uint256 _usdcToRepay, uint256 _wethToWithdraw) external {
```

Comment:
confusing with _repayAndWithdrawFromLeverage

Reply:
yeah agreed

Reply:
renamed to ```auctionRepayAndWithdrawFromLeverage()```